### ISSUE TITLE
Allow self-signed certificate in electron app

### ISSUE THREAD

**Issue by VigneshVSV** [VigneshVSV/thing-control-panel#2]
The electron build is stored in a different repository [here](https://gitlab.com/hololinked-server/thing-control-panel-electron-build)

It does not allow self signed certificates (and also probably unencrypted HTTP). Possible steps
- [ ] add logic that allows this in a safe way. User should try to open in new tab and allow certificate, just like a normal browser.
- [ ] open in new tab not working, so fix that as well
- [ ] Error message when certificate is not working is shown as "Network Error", improve that to suggest at least that the cause of the error might be the certificate.

**Comment by samubura** [comment]
I was trying to use the app with a self-hosted node-wot TD in localhost over HTTP 
the error I get is related to [mixed content](https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content?utm_source=mozilla&utm_medium=firefox-console-errors&utm_campaign=default)
`Blocked loading mixed active content “http://192.168.1.111:8080/thing”`

Maybe allowing the app to be run in HTTP instead of HTTPS with a flag might already fix this?
I was running using npm run dev

**Comment by VigneshVSV** [comment]
Yes I think the issue is I forgot to add a `HttpClientFactory` apart from `HttpsClientFactory` within the node-wot client factories. 

I am already testing, I will keep you updated in 10 minutes.

**Comment by VigneshVSV** [comment]
I was wrong, you are right.

I just need to add a flag that can run the app with HTTP instead of HTTPS. For time being, you could comment out `mkcert` plugin in vite.config.ts line 10.

**Comment by VigneshVSV** [comment]
There is a small config file in 'src\config.tsx' that toggles HTTP(s), you could try it out. I will late move it to the top level when some details are correct.

**Comment by VigneshVSV** [comment]
Hi @samubura I just deployed the app as a website directly to get around these issues.
SSL here - https://control-panel.hololinked.dev/
no SSL here - http://no-ssl-control-panel.hololinked.net/

Funnily, I never knew about this technicality.
